src/process/executable_path.cr

```diff
@@ -90,6 +90,24 @@ end
       "/proc/self/exe"
     end
   end
+{% elsif flag?(:win32) %}
+  require "crystal/system/windows"
+  require "c/libloaderapi"
+
+  class Process
+    private def self.executable_path_impl
+      Crystal::System.retry_wstr_buffer do |buffer, small_buf|
+        len = LibC.GetModuleFileNameW(nil, buffer, buffer.size)
+        if 0 < len < buffer.size
+          break String.from_utf16(buffer[0, len])
+        elsif small_buf && len == buffer.size
+          next 32767 # big enough. 32767 is the maximum total path length of UNC path.
+        else
+          raise ::File::Error.from_winerror("Error getting executable path", file: "?")
```

Comment:
Wait, what does `file: "?"` mean?

Reply:
I should have heard opinions about it first.

At first I wrote the line without `file: "?"` and got the following error.
```
In src/file/error.cr:11:27

 11 | File::NotFoundError.new(message, **opts)
                          ^--
Error: missing argument: file
```
I came up with two ideas. One is using another error class. The other is supplying a dummy file name such as `"?"`. I thought that the error was related to file so the `File::Error` was proper. So I chose the latter. However I have not been satisfied with it yet.

Any idea in order not to use `file: "?"`?

Reply:
Hm well why is is `::File::Error` in the first place? That is what requires a `file` arg.

BTW I don't see any other `executable_path_impl` raising any errors, they just return `nil`. Maybe it should be that?

If not, maybe go back to `RuntimeError` and possibly specialize the exception in a different direction?

Reply:
It should just return `nil` if the file can't be found, as per https://crystal-lang.org/api/0.34.0/Process.html#executable_path-class-method

Reply:
@straight-shoota 
Thanks. I fixed it just to return `nil` on error.